scarlet/psf.py

```diff
@@ -152,3 +156,125 @@ def update_dtype(self, dtype):
         if self.image.dtype != dtype:
             self._image = self._image.astype(dtype)
         return self
+
+
+class PSFDiffKernel(Component):
+    """PSF Difference Kernel Source
+
+    "Source" used to model the difference kernel to match one
+    PSF to another. Typically this should be modeling the
+    deconvolution kernel to convolve an image from an
+    observed PSF into a model PSF, as the difference
+    from model PSF to observed PSF is calculated much
+    more quickly using a DFT in `~scarlet.Observation.match`.
+    """
+    def __init__(
+        self,
+        frame,
+        initial,
+        band,
+        step=1e-2
+    ):
+        """Initialize the Model
+
+        Parameters
+        ----------
+        frame: `~scarlet.Frame`
+            The spectral and spatial characteristics of this component.
+        initial: `numpy.array`
+            Initial guess for the kernel.
+        step: `double`
+            Step size for the kernel parameter.
+        """
+        kernel = np.zeros(initial.shape, dtype=initial.dtype)
+        kernel[band] = initial[band]
+        kernel = Parameter(
+            kernel,
+            name="kernel",
+            step=step,
+        )
+        super().__init__(frame, kernel)
+
+    def get_model(self, *parameters):
+        kernel = self.kernel
+        if len(parameters) == 1:
+            kernel = parameters[0]
+        elif len(parameters) > 1:
+            raise ValueError("PsfDiffKernel only takes a single parameter")
+        return kernel
+
+    @property
+    def kernel(self):
+        """Return the contents of the kernel parameter
+        """
+        return self._parameters[0]._data
+
+
+class PsfObservation(Observation):
+    def match(self, psfs, convolution = "real"):
+        """Implement the observed PSFs as the difference kernel
+
+        This is different than `~scarlet.Observation.match`,
+        where the difference kernel that matches the model
+        PSF to the observed PSF is calculated and stored.
+        For a `PsfObservation` the input PSF, which
+        should be the observed PSF (see below) is stored
+        as the "difference kernel" while the actual
+        deconvolution (difference) kernel is calculated
+        by the model.
```

Comment:
I am not sure what this is for. Could you please add in the description what this achieves? My impression is that this will allow full deconvolution, but I don't why.

Reply:
This is part of the deconvolution PR (see #161 ), which this branch is being merged into. So please see that PR and comment there if you still feel it is unclear in that context.